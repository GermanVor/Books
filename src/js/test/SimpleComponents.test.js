import React from 'react'
import Enzyme, { shallow, mount, render } from 'enzyme'
import Paggination from '../components/Pagination'
import renderer from 'react-test-renderer';

describe('<Paggination /> ', () => {
  it("render correctly Paggination component ", () => {
    const Pag = renderer.create(<Paggination DBSize={7} />).toJSON();
    expect(Pag).toMatchSnapshot();
  });

  it("check the type of Paggination value ", () => {
    const DBSize =100, limit = 5, page = 10;
    const Pag = mount(<Paggination  
      DBSize = { DBSize } 
      onClick = { ()=>true }
      limit = { limit }
      page = { page }
    />);
    expect(Pag.prop("DBSize")).toBe(DBSize);
    expect(Pag.prop("limit")).toBe(limit);
    expect(Pag.prop("page")).toBe(page);
    expect(Pag.find(".btn-group button")).toHaveLength(Math.ceil(DBSize/limit));
    expect(Pag.find(".btn-group button[page="+page+"].activPagin")).toHaveLength(1);
    expect(Pag.find(".LimitMenu button[limit='"+limit+"'].activPagin")).toHaveLength(1);
  });

  it("check the resize operation ", () => {
    const DBSize = 100, limit = 5, page = 10;
    const Pag = mount(<Paggination  
      DBSize = { DBSize } 
      onClick = { ()=>true }
      limit = { limit }
      page = { page }
    />);
    expect(Pag.find(".btn-group button")).toHaveLength(Math.ceil(DBSize/limit));

    Pag.find('.LimitMenu button[limit="'+3+'"]').simulate('click');
    expect(Pag.find(".btn-group button")).toHaveLength(Math.ceil(DBSize/3));

    Pag.find('.LimitMenu button[limit="'+5+'"]').simulate('click');
    expect(Pag.find(".btn-group button")).toHaveLength(Math.ceil(DBSize/5));
 
    Pag.find('.LimitMenu button[limit="'+10+'"]').simulate('click');
    expect(Pag.find(".btn-group button")).toHaveLength(Math.ceil(DBSize/10));
    
  });

});

