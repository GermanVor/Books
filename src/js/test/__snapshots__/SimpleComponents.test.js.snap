// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paggination />  render correctly Paggination component  1`] = `
<div
  className="Pagination"
>
  <div
    className="LimitMenu"
  >
    <button
      className="btn btn-info activPagin"
      limit={3}
      onClick={[Function]}
    >
      3
    </button>
    <button
      className="btn btn-info "
      limit={5}
      onClick={[Function]}
    >
      5
    </button>
  </div>
  <div
    aria-label="First group"
    className="btn-group mr-2"
    role="group"
  >
    <button
      className="btn btn-info activPagin"
      onClick={[Function]}
      page={0}
      type="button"
    >
      0
    </button>
    <button
      className="btn btn-info "
      onClick={[Function]}
      page={1}
      type="button"
    >
      1
    </button>
    <button
      className="btn btn-info "
      onClick={[Function]}
      page={2}
      type="button"
    >
      2
    </button>
  </div>
</div>
`;
